source/MRMesh/MRVoxelsConversions.cpp

```diff
@@ -474,74 +489,185 @@ TriangulationPlan{0, 3, 8},
 TriangulationPlan{}
 };
 
+// check if this iter is needed for config
+bool cNeedIter( int iterIndex, uint8_t config )
+{
+    if ( iterIndex == 0 )
+    {
+        auto base = ( config & ( 1 << 0 ) );
+        if ( base != ( config & ( 1 << cMapNeighborsShift[1] ) ) )
```

Comment:
Can we prepare `( 1 << cMapNeighborsShift[1] )` beforehand to be sure it is not computed in runtime?